pilot/pkg/ambient/ambient.go

```diff
@@ -17,13 +17,16 @@ package ambient
 type NodeType = string
 
 const (
-	LabelStatus = "istio.io/ambient-status"
-	TypeEnabled = "enabled"
-	// LabelType == "workload" -> intercept into ztunnel
-	// TODO this could be an annotation – eventually move it into api repo
-	LabelType = "ambient-type"
-
-	TypeWorkload NodeType = "workload"
-	TypeNone     NodeType = "none"
+	// AnnotationType determines the type of workload this is.
+	AnnotationType = "ambient.istio.io/type"
+	// TypeMesh indicates this workload is in the mesh and has redirection to ztunnel.
+	// This is added by the CNI when redirection is configured, not by the user.
+	TypeMesh NodeType = "mesh"
```

Comment:
mesh could also indicate sidecar too, how about `sidecarless` as the the workload has redirection to ztunnel?

Reply:
It's already prefixed with `ambient.istio.io` which should be enough to indicate the scope here is only ambient.

Reply:
yeah, except we are having discussion in ambient-dev that ambient also includes sidecar as well.

Reply:
We can easily bikeshed the naming later, I hope we will not block development on naming. Its trivial to find+replace it before we ship if we decide to.

Reply:
That is fair - i agree this is a bigger discussion than this PR itself

Reply:
Something close to what John already suggested: `ambient.istio.io/redirection: {node,sidecar}` might do.